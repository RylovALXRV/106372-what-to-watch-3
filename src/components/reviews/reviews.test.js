import React from "react";
import renderer from "react-test-renderer";
import Reviews from "./reviews.jsx";

const mock = {
  director: `David Yates`,
  starring: [`Rami Malek`, `Jon Finch`, `Cate Blanchett`],
  duration: `1h 30m`,
  genre: `drama`,
  year: `2018`,
  reviews: [{
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: 8.0,
    author: `Kate Muir`,
    date: `December 20, 2016`,
  }],
  descriptions: `In late 1823, Hugh Glass guides Captain Andrew Henry's trappers through territory of the present day Dakotas.`,
  rating: 8.0,
};

it(`Should Reviews render correctly`, () => {
  const tree = renderer.create(
      <Reviews
        movie={mock}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
