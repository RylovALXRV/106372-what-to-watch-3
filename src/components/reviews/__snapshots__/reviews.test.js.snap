// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Reviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Kate Muir
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-20"
          >
            December 20, 2016
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
